### PR TITLE
ARCH-252 Added github workflow for tech docs copier

### PR DESCRIPTION
**Pull request by glothriel**

### CODE REVIEW

**Comment by glothriel** [comment]
I need to add a readme to that, but please do a first round of review. It can be used on any repo, just add a workflow as follows:

```
name: Documentation
on: [push]

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

jobs:
  push:
  
    uses: PiwikPRO/actions/.github/workflows/push_docs.yaml@arch-252-gh-workflow-for-tech-docs-copier
    secrets: inherit
    with:
      config: |
        {
          "documents": [
            {
              "source": "docs/*",
              "destination": "barman/",
              "project": "promil"
            }
          ]
        }

```

and adjust the config

File: .github/workflows/push_docs.yaml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+      - name: Create Branch if Not Exists
+        if: github.ref_name != env.MAIN_BRANCH && env.BRANCH_EXISTS == 'false'
+        run: |
+          cd Tech-docs
+          git checkout -b $TECH_DOCS_BRANCH
+          echo "CHANGES=1" >> $GITHUB_ENV
+
+      - name: Run Copy Script
+        uses: PiwikPRO/actions/techdocs@arch-252-gh-workflow-for-tech-docs-copier
```
Comment (glothriel) [inline .github/workflows/push_docs.yaml:109]: must be changed before merge

File: README.md [hunk 2]
```diff
@@ -510,3 +514,43 @@ making it as generic as possible.
...
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  push:
+
+    uses: PiwikPRO/actions/.github/workflows/push_docs.yaml@master
+    secrets: inherit
+    with:
+      config: |
```
Comment (jzyinq) [inline README.md:538]: Maybe we should drop some hint about this:
```yaml
main_branch:
  default: master
  required: false
  type: string
```

We don't mention this in README, which could be valuable in some cases.

File: .github/workflows/push_docs.yaml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+          repository: 'PiwikPRO/Tech-docs'
+          token: ${{ steps.get-token.outputs.token }}
+          path: 'Tech-docs'
+
+      - name: Set Tech-docs branch name
+        if: github.ref_name != env.MAIN_BRANCH
+        shell: bash
+        run: |
+          CURRENT_BRANCH="${{ github.ref_name }}"
+          BRANCH=$(echo '${{ github.repository }}-'$CURRENT_BRANCH | sed 's/[^a-zA-Z0-9]/-/g; y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/')
```
Comment (jzyinq) [inline .github/workflows/push_docs.yaml:47]:
```suggestion
          BRANCH=$(echo '${{ github.repository }}-'$CURRENT_BRANCH | sed 's/[^a-zA-Z0-9]/-/g; s/[A-Z]/\L&/g')
```

Comment (jzyinq) [reply .github/workflows/push_docs.yaml:47]:
```suggestion
          BRANCH=$(echo '${{ github.repository }}-'$CURRENT_BRANCH | 's/[^a-zA-Z0-9]/-/g;' | tr '[:upper:]' '[:lower:]'])
```

Comment (jzyinq) [reply .github/workflows/push_docs.yaml:47]: apples, oranges, choose your favorite

File: .github/workflows/push_docs.yaml [hunk 4]
```diff
@@ -0,0 +1,126 @@
+name: Push the documentation to Tech-docs
```
Comment (jzyinq) [inline .github/workflows/push_docs.yaml:1]:
```suggestion
name: 'Tech-docs copier'
description: 'This should not be used directly by end-users, please use the reusable workflow instead.'
```

File: .github/workflows/push_docs.yaml [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+          index: ${{ github.repository }}
+          dry-run: false
+          config: ${{ inputs.config }}
+
+      - name: Push changes to Tech-docs
+        if: env.CHANGES != '0'
+        run: |
+          cd Tech-docs
+          git config --global user.email "${{github.event.pusher.email}}"
+          git config --global user.name "${{github.event.pusher.name}}"
```
Comment (jzyinq) [inline .github/workflows/push_docs.yaml:123]: nice

File: .github/workflows/push_docs.yaml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+          private-key: ${{ secrets.TECH_DOCS_PUSHER_PRIVATE_KEY }}
+          app-id: ${{ secrets.TECH_DOCS_PUSHER_APPLICATION_ID }}
+          repo: PiwikPRO/Tech-docs
+
+      - name: Check out Tech-docs
+        uses: actions/checkout@v3
+        with:
+          repository: 'PiwikPRO/Tech-docs'
+          token: ${{ steps.get-token.outputs.token }}
+          path: 'Tech-docs'
```
Comment (jzyinq) [inline .github/workflows/push_docs.yaml:40]: can we make `Tech-docs` an env? We have a lot of occurrences of this.

File: techdocs/action.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    required: true
+  config:
+    required: true
+  dry-run:
+    required: false
+    default: 'false'
+outputs:
+  changes:
+    description: "Textual description of changes"
+    value: ${{ steps.dry-run.outputs.changes }}
```
Comment (jzyinq) [inline techdocs/action.yaml:17]: Where this output is used?

Comment (glothriel) [reply techdocs/action.yaml:17]: In next PR :D

File: .github/workflows/push_docs.yaml (hunk 6 above)
Comment (jzyinq) [reply .github/workflows/push_docs.yaml:41]: What about this?

Comment (glothriel) [reply .github/workflows/push_docs.yaml:41]: I hoped you forgot about this :D

Comment (jzyinq) [reply .github/workflows/push_docs.yaml:41]: As we discussed - not worth it

**Review by jzyinq — APPROVED** [review]

**Pull request merged**